src/Microsoft.Diagnostic.TestHelpers/TestConfiguration.cs

```diff
@@ -559,11 +559,12 @@ public bool CreateDumpExists
         }
 
         /// <summary>
-        /// Returns true if a stack overflow causes dump to be generated with createdump. Currently no .NET Core version does.
+        /// Returns true if a stack overflow causes dump to be generated with createdump. 3.x has now started to
+        /// create dumps on stack overflow.
         /// </summary>
         public bool StackOverflowCreatesDump
         {
-            get { return false; }
+            get { return IsNETCore && RuntimeFrameworkVersion.StartsWith("3."); }
```

Comment:
You want to handle versions >3 while you are at it, I'm sure it won't be too far in the future when that happens.